test/Graphics/Shape.cpp

```diff
@@ -0,0 +1,89 @@
+#include <SFML/Graphics/Shape.hpp>
+#include "GraphicsUtil.hpp"
+
+#include <doctest.h>
+
+using doctest::Approx;
+
+class TriangleShape : public sf::Shape
+{
+public:
+    explicit TriangleShape(const sf::Vector2f& size);
+    std::size_t getPointCount() const override;
+    sf::Vector2f getPoint(std::size_t index) const override;
+private:
+    sf::Vector2f m_size;
+};
+
+TEST_CASE("sf::Shape class - [graphics]")
+{
+    SUBCASE("Default constructor")
+    {
+        const TriangleShape triangleShape({0, 0});
+        CHECK(triangleShape.getTexture() == nullptr);
+        CHECK(triangleShape.getTextureRect() == sf::IntRect());
+        CHECK(triangleShape.getFillColor() == sf::Color::White);
+        CHECK(triangleShape.getOutlineColor() == sf::Color::White);
+        CHECK(triangleShape.getOutlineThickness() == 0.0f);
+        CHECK(triangleShape.getLocalBounds() == sf::FloatRect());
+        CHECK(triangleShape.getGlobalBounds() == sf::FloatRect());
+    }
+
+    SUBCASE("Setters and getters")
+    {
+        TriangleShape triangleShape({0, 0});
+        triangleShape.setTextureRect({{4, 5}, {6, 7}});
+        triangleShape.setFillColor(sf::Color::Cyan);
+        triangleShape.setOutlineColor(sf::Color::Magenta);
+        triangleShape.setOutlineThickness(3.14f);
+        CHECK(triangleShape.getTextureRect() == sf::IntRect({4, 5}, {6, 7}));
+        CHECK(triangleShape.getFillColor() == sf::Color::Cyan);
+        CHECK(triangleShape.getOutlineColor() == sf::Color::Magenta);
+        CHECK(triangleShape.getOutlineThickness() == 3.14f);
+    }
+
+    SUBCASE("Virtual functions")
+    {
+        const TriangleShape triangleShape({2, 2});
+        CHECK(triangleShape.getPointCount() == 3);
+        CHECK(triangleShape.getPoint(0) == sf::Vector2f(1, 0));
+        CHECK(triangleShape.getPoint(1) == sf::Vector2f(0, 2));
+        CHECK(triangleShape.getPoint(2) == sf::Vector2f(2, 2));
+    }
+
+    SUBCASE("Get bounds")
+    {
+        TriangleShape triangleShape({2, 3});
+        CHECK(triangleShape.getLocalBounds() == sf::FloatRect({0, 0}, {2, 3}));
+        CHECK(triangleShape.getGlobalBounds() == sf::FloatRect({0, 0}, {2, 3}));
+        triangleShape.move({1, 1});
+        triangleShape.rotate(sf::degrees(90));
+        CHECK(triangleShape.getLocalBounds() == sf::FloatRect({0, 0}, {2, 3}));
+        CHECK(triangleShape.getGlobalBounds().left == Approx(-2.f));
+        CHECK(triangleShape.getGlobalBounds().top == Approx(1.f));
+        CHECK(triangleShape.getGlobalBounds().width == Approx(3.f));
+        CHECK(triangleShape.getGlobalBounds().height == Approx(2.f));
```

Comment:
I suppose it's testing both. To ensure that `getGlobalBounds` is still correct after moving an object, I had to move the object. Otherwise we're only testing the initial global bounds. In theory that function could be bugged and never update the global bounds after being moved. This test ensures that is not the case.